lib/get-config.js

```diff
@@ -303,7 +310,60 @@ function validateRules(config, options) {
   }
 }
 
-module.exports = function(options) {
+function _determineConfigForSeverity(config) {
+  switch (config) {
+    case SEVERITY_LEVEL_IGNORE:
+      return { config: false, severity: IGNORE_SEVERITY };
+    case SEVERITY_LEVEL_WARNING:
+      return { config: true, severity: WARNING_SEVERITY };
+    case SEVERITY_LEVEL_ERROR:
+      return { config: true, severity: ERROR_SEVERITY };
+  }
+}
+
+/**
+ * Sets the severity and config on the rule object.
+ * Eg:
+ * {'no-implicit-this': 'warn'} -> {'no-implicit-this': { severity: 'warn', config: true }}
+ * { 'no-implicit-this': [ 'warn', { allow: [ 'fooData' ] } ] }
+ * would become:
+ * { 'no-implicit-this': { severity: 'warn', config: 'allow': [ 'fooData' ] } }
+ * @param {*} configData
+ */
+function processRules(config) {
+  for (let key in config.rules) {
+    let severityConfig;
+    let ruleData = config.rules[key];
+    let ruleConfig = {};
+    // In case of {'no-implicit-this': 'off|warn|error'}
+    if (typeof ruleData === 'string') {
+      severityConfig = _determineConfigForSeverity(ruleData);
+      if (severityConfig) {
+        ruleConfig.severity = severityConfig.severity;
+        ruleConfig.config = severityConfig.config;
+        config.rules[key] = ruleConfig;
+      }
+    }
+    // array of severity and custom rule config
+    else if (Array.isArray(ruleData)) {
+      if (ruleData.length !== 2) {
+        throw new Error(
+          'Severity and custom rule config are the only two values allowed in the rule config array.'
+        );
+      }
```

Comment:
I don't think we can throw this error. Rules are allowed to accept an array as their configuration today, adding this error would be a breaking change.

Instead we need to check if it is an array **and** if the first value is one of our known strings (warn, error, off). If the first value is _not_ one of our strings, we need to fallback to current behavior (not error).

Reply:
From the offline discussion with @rwjblue , our eventual goal is to get devs to follow the standard pattern when arrays are used (first severity and then the custom config), however, this would be a breaking change and we would want to gradually transition the devs towards this pattern.
Will change it fallback to the default implementation